skimage/feature/_hog.py

```diff
@@ -1,37 +1,68 @@
 from __future__ import division
 import numpy as np
 from .._shared.utils import assert_nD
+from .._shared.utils import skimage_deprecation, warn
 from . import _hoghistogram
-import warnings
 
 
-def hog(image, orientations=9, pixels_per_cell=(8, 8),
-        cells_per_block=(3, 3), visualise=False, transform_sqrt=False,
+def _hog_normalize_block(block, method, eps=1e-5):
+    if method == 'L1':
+        out = block / np.sqrt(block.sum() ** 2 + eps)
+    elif method == 'L1-sqrt':
+        out = np.sqrt(block / np.sqrt(block.sum() ** 2 + eps))
+    elif method == 'L2':
+        out = block / np.sqrt((block ** 2).sum() + eps ** 2)
+    elif method == 'L2-Hys':
+        out = block / np.sqrt((block ** 2).sum() + eps ** 2)
+        out = np.minimum(out, 0.2)
+        out = out / np.sqrt((out ** 2).sum() + eps ** 2)
+    else:
+        raise ValueError('Selected block normalization method is invalid.')
+
+    return out
+
+
+def hog(image, orientations=9, pixels_per_cell=(8, 8), cells_per_block=(3, 3),
+        block_norm='L1', visualise=False, transform_sqrt=False,
         feature_vector=True, normalise=None):
     """Extract Histogram of Oriented Gradients (HOG) for a given image.
 
     Compute a Histogram of Oriented Gradients (HOG) by
 
-        1. (optional) global image normalisation
+        1. (optional) global image normalization
         2. computing the gradient image in x and y
         3. computing gradient histograms
-        4. normalising across blocks
+        4. normalizing across blocks
         5. flattening into a feature vector
 
     Parameters
     ----------
     image : (M, N) ndarray
         Input image (greyscale).
-    orientations : int
+    orientations : int, optional
         Number of orientation bins.
-    pixels_per_cell : 2 tuple (int, int)
+    pixels_per_cell : 2-tuple (int, int), optional
         Size (in pixels) of a cell.
-    cells_per_block  : 2 tuple (int,int)
+    cells_per_block : 2-tuple (int, int), optional
         Number of cells in each block.
+    block_norm : str {'L1', 'L1-sqrt', 'L2', 'L2-Hys'}, optional
+        Block normalization method:
+
+        ``L1``
+           Normalization using L1-norm. (default)
+        ``L1-sqrt``
+           Normalization using L1-norm, followed by square root.
+        ``L2``
+           Normalization using L2-norm.
+        ``L2-Hys``
```

Comment:
I'd prefer to stick to the capitalized version, as both articles use the very same notation.